src/pages/storage/panels/CreateBucketPanel.tsx

```diff
@@ -0,0 +1,131 @@
+import {
+  ActionButton,
+  Button,
+  useNotify,
+  useToastNotification,
+} from "@canonical/react-components";
+import SidePanel from "components/SidePanel";
+import type { FC } from "react";
+import { useState } from "react";
+import usePanelParams from "util/usePanelParams";
+import * as Yup from "yup";
+import { useFormik } from "formik";
+import NotificationRow from "components/NotificationRow";
+import ScrollableContainer from "components/ScrollableContainer";
+import ResourceLink from "components/ResourceLink";
+import StorageBucketForm from "../forms/StorageBucketForm";
+import type { StorageBucketFormValues } from "../forms/StorageBucketForm";
+import { createStorageBucket } from "api/storage-buckets";
+import { queryKeys } from "util/queryKeys";
+import { useQueryClient } from "@tanstack/react-query";
+import { testDuplicateStorageBucketName } from "util/storageBucket";
+
+const CreateBucketPanel: FC = () => {
+  const panelParams = usePanelParams();
+  const notify = useNotify();
+  const toastNotify = useToastNotification();
+  const controllerState = useState<AbortController | null>(null);
+  const queryClient = useQueryClient();
+  const closePanel = () => {
+    panelParams.clear();
+    notify.clear();
+  };
+
+  const bucketSchema = Yup.object().shape({
+    name: Yup.string()
+      .test(
+        ...testDuplicateStorageBucketName(panelParams.project, controllerState),
+      )
+      .required("Bucket name is required"),
+  });
+
+  const handleSuccess = (bucketName: string) => {
+    toastNotify.success(
+      <>
+        Bucket{" "}
+        <ResourceLink
+          type="volume"
+          value={bucketName}
+          to={`/ui/${panelParams.project}/default/storage/buckets`}
+        />{" "}
+        created.
+      </>,
+    );
+    closePanel();
+  };
+  const formik = useFormik<StorageBucketFormValues>({
+    initialValues: {
+      name: "",
+      pool: "",
+      size: "",
+      description: "",
+      target: "",
+    },
+    validationSchema: bucketSchema,
+    onSubmit: (values) => {
+      const bucketPayload = {
+        name: values.name,
+        config: { size: values.size },
+        description: values.description,
```

Comment:
We are sending both parameters size and description, even if they are not set:

![image](https://github.com/user-attachments/assets/e44535b7-f8f7-45c7-8753-de6d97fd93ef)

We should avoid setting them in case the value is empty. I think we need to adjust the bucket type accordingly as the description is then optional.